nixos/modules/hardware/hackrf.nix

```diff
@@ -0,0 +1,19 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.hardware.hackrf;
+
+in
+{
+  options.hardware.hackrf = {
+    enable = lib.mkEnableOption ''
+      Enables hackrf udev rules and ensures 'plugdev' group exists.
+      This is a prerequisite to using HackRF devices without being root, since HackRF USB descriptors will be owned by plugdev through udev
+    '';
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.udev.packages = [ pkgs.hackrf ];
+    users.groups.plugdev = { };
```

Comment:
@andir was it you who has thoughts about the `plugdev` group?

Reply:
Not really. My opinion on that group is that it is really too wide and there are no good reasons to use it other than that everyone hardcoded it in their udev rules :(

Reply:
I also thought it was broader than ideal at first, but it is the same group as used in the `rtl-sdr` module (and for the same reason). I figured using the same group made sense since both are for SDR hardware.